docker-compose.yaml

```diff
@@ -0,0 +1,142 @@
+            - ORGANIZATION_ADMIN_EMAIL={my_organization_web_admin_email}
+            - DRAMATIQ_BROKER_HOST=redis
+            - DRAMATIQ_BROKER_PORT=6379
+            - MAX_FAILURES_PER_REQUEST=3
+            - CORS_ALLOW_CREDENTIALS=true
+            - CORS_ORIGIN_WHITELIST=http://localhost:8080,http://127.0.0.1:8080
+            - CSRF_TRUSTED_ORIGINS=localhost:8080,127.0.0.1:8080
+        mem_limit: "512m"
+        restart: always
+        volumes:
+            - static_volume:/static
+        command: >
+            sh -c "python manage.py migrate
+            && python manage.py collectstatic --no-input
+            && gunicorn --bind=0.0.0.0:8001 --worker-class=gevent --workers=4 --access-logfile=- --error-logfile=- observation_portal.wsgi:application --timeout=300"
+        depends_on:
+            redis:
+                condition: service_healthy
+
+    dramatiq_task_scheduler:
+        image: observatorycontrolsystem/observation-portal:3.5.4
+        environment:
+            - DB_HOST=172.17.0.1
+            - DB_PORT=5432
+            - DB_NAME=observationportal
+            - DB_USER=postgres
+            - DB_PASSWORD=postgres
+            - SECRET_KEY=obs_portal_secret_key
+            - CONFIGDB_URL=http://172.17.0.1:7000
+            - DOWNTIMEDB_URL=http://172.17.0.1:7500
+            - ELASTICSEARCH_URL=
+            - CORS_ALLOW_CREDENTIALS=true
+            - CORS_ORIGIN_WHITELIST=http://localhost:8080,http://127.0.0.1:8080
+            - CSRF_TRUSTED_ORIGINS=localhost:8080,127.0.0.1:8080
+            - DRAMATIQ_BROKER_HOST=redis
+            - DRAMATIQ_BROKER_PORT=6379
+        mem_limit: "512m"
+        restart: always
+        command: >
+            sh -c "python manage.py runscript observation_portal.task_scheduler"
+        depends_on:
+            redis:
+                condition: service_healthy
+
+    dramatiq_worker:
+        image: observatorycontrolsystem/observation-portal:3.5.4
+        environment:
+            - DB_HOST=172.17.0.1
+            - DB_PORT=5432
+            - DB_NAME=observationportal
+            - DB_USER=postgres
+            - DB_PASSWORD=postgres
+            - DEBUG=true
+            - SECRET_KEY=obs_portal_secret_key
+            - CONFIGDB_URL=http://172.17.0.1:7000
+            - DOWNTIMEDB_URL=http://172.17.0.1:7500
+            - ELASTICSEARCH_URL=
+            - CORS_ALLOW_CREDENTIALS=true
+            - CORS_ORIGIN_WHITELIST=http://localhost:8080,http://127.0.0.1:8080
+            - CSRF_TRUSTED_ORIGINS=localhost:8080,127.0.0.1:8080
+            - DRAMATIQ_BROKER_HOST=redis
+            - DRAMATIQ_BROKER_PORT=6379
+        mem_limit: "512m"
+        restart: always
+        command: >
+            sh -c "python manage.py rundramatiq --processes 2 --threads 4"
+        depends_on:
+            redis:
+                condition: service_healthy
+
+    # This provides a simple example frontend that can view and schedule Requests using
```

Comment:
I'm on the fence about whether to include the frontend in here. Originally I was thinking we would just have the docker compose file for each repo contain the services necessary for running the app in that repo. I guess since for a prod deploy you'd run the frontend separately and the frontend would probably be different from the example frontend, given that this is a base for people to work from when starting their projects in prod, I'm leaning towards saying remove it from here and instead have a separate compose file in the frontend repo that runs just the frontend. What do you think?

Reply:
hmm yeah I guess that can work... I'm not sure how to setup the nginx (or two nginxes?) then to route /app and /o and /admin stuff to the backend but everything else to the frontend. Maybe we just only have an nginx on the backend and route any non-backend endpoints to the frontend ip, but that ip would need to be changed from default in the nginx conf if they run the frontend somewhere else...

Reply:
Right... I think it might just work since we do it in the ocs_example stack without adding an extra nginx there, but I'm not 100% sure.

Reply:
in the ocs_example stack, we have the api and frontend running from different ports, so no nginx is needed (and debug=True is on to host the staticfiles). But I'll see if I can work something out...

Reply:
nevermind... i didn't realize the ocs_example_frontend bakes in nginx into its dockerfile, and then uses that to route to the backend endpoints with the environment variable set. It all works fine separately